prove_withdrawal.go

```diff
@@ -166,153 +160,141 @@ func (epp *EigenPodProofs) ProveWithdrawals(
 
-	// setup the withdrawal block header
 	withdrawalBlockHeader := &phase0.BeaconBlockHeader{
-		Slot:          withdrawalBlock.Slot,
-		ProposerIndex: withdrawalBlock.ProposerIndex,
-		ParentRoot:    withdrawalBlock.ParentRoot,
-		StateRoot:     withdrawalBlock.StateRoot,
+		Slot:          slot,
+		ProposerIndex: proposerIndex,
+		ParentRoot:    parentRoot,
+		StateRoot:     stateRoot,
 		BodyRoot:      blockBodyRoot,
 	}
 
 	// initialize the withdrawal proof
 	withdrawalProof := &WithdrawalProof{}
 
 	start = time.Now()
-	// prove the execution payload against the withdrawal block header
-	withdrawalProof.ExecutionPayloadProof, withdrawalProof.ExecutionPayloadRoot, err = beacon.ProveExecutionPayloadAgainstBlockHeaderCapella(withdrawalBlockHeader, withdrawalBlock.Body)
-	if err != nil {
-		return nil, err
+	var withdrawalExecutionPayloadFieldRoots []phase0.Root
+	var withdrawals []*capella.Withdrawal
+	var withdrawalFields []Bytes32
+	var withdrawalIndex uint64
+	var timestamp uint64
+	if withdrawalBlock.Version == spec.DataVersionDeneb {
+		start = time.Now()
+		// prove the execution payload against the withdrawal block header
+		withdrawalProof.ExecutionPayloadProof, withdrawalProof.ExecutionPayloadRoot, err = beacon.ProveExecutionPayloadAgainstBlockHeaderDeneb(withdrawalBlockHeader, withdrawalBlock.Deneb.Body)
+		if err != nil {
+			return nil, nil, err
+		}
+		log.Info().Msgf("time to prove execution payload against block header: %s", time.Since(start))
+
+		// calculate execution payload field roots
+		withdrawalExecutionPayloadFieldRoots, err = beacon.ComputeExecutionPayloadFieldRootsDeneb(withdrawalBlock.Deneb.Body.ExecutionPayload)
+		if err != nil {
+			return nil, nil, err
+		}
+		withdrawals = withdrawalBlock.Deneb.Body.ExecutionPayload.Withdrawals
+		withdrawalIndex = GetWithdrawalIndex(validatorIndex, withdrawals)
+		withdrawalFields = ConvertWithdrawalToWithdrawalFields(withdrawalBlock.Deneb.Body.ExecutionPayload.Withdrawals[withdrawalIndex])
+		timestamp = withdrawalBlock.Deneb.Body.ExecutionPayload.Timestamp
+	} else if withdrawalBlock.Version == spec.DataVersionCapella {
+		start = time.Now()
+		// prove the execution payload against the withdrawal block header
+		withdrawalProof.ExecutionPayloadProof, withdrawalProof.ExecutionPayloadRoot, err = beacon.ProveExecutionPayloadAgainstBlockHeaderCapella(withdrawalBlockHeader, withdrawalBlock.Capella.Body)
+		if err != nil {
+			return nil, nil, err
+		}
+		log.Info().Msgf("time to prove execution payload against block header: %s", time.Since(start))
+
+		// calculate execution payload field roots
+		withdrawalExecutionPayloadFieldRoots, err = beacon.ComputeExecutionPayloadFieldRootsCapella(withdrawalBlock.Capella.Body.ExecutionPayload)
+		if err != nil {
+			return nil, nil, err
+		}
+		withdrawals = withdrawalBlock.Capella.Body.ExecutionPayload.Withdrawals
+		withdrawalIndex = GetWithdrawalIndex(validatorIndex, withdrawals)
```

Comment:
these are done in both branches, take outside of if/else block and put under

Reply:
problem is that its withdrawalBlock.Capella.Body.ExecutionPayload.Withdrawals vs withdrawalBlock.Deneb.Body.ExecutionPayload.Withdrawals in the other.

Reply:
you can keep `withdrawals = ...` just move the other stuff